src/transformers/models/bart/modeling_bart.py

```diff
@@ -294,6 +295,14 @@ class BartFlashAttention2(BartAttention):
     flash attention and deal with padding tokens in case the input contains any of them.
     """
 
+    # Copied from transformers.models.llama.modeling_llama.LlamaFlashAttention2.__init__
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # flash_attn<2.1 generates top-left aligned causal mask, while what is needed here is bottom-right alignement, that was made default for flash_attn>=2.1. This attribute is used
+        # to handle this difference.
+        self.flash_attn_uses_top_left_mask = not is_flash_attn_greater_or_equal_210()
```

Comment:
```suggestion
        self._flash_attn_uses_top_left_mask = not is_flash_attn_greater_or_equal_210()
```
I would make this attribute private for all models, making it public might lead users to customly use that variable for other purposes, and in case we change our mind in the future for some reason we will still need to keep it for BC reasons (happened for `gradient_checkpointing` attribute in the past)

Reply:
Fixed makes sense!